packages/node_modules/@webex/plugin-meetings/src/reconnection-manager/index.js

```diff
@@ -41,6 +54,50 @@ export default class ReconnectionManager {
     this.configure(meeting);
   }
 
+  /**
+   * @returns {undefined}
+   * @public
+   * @memberof ReconnectionManager
+   */
+  iceReconnected() {
+    if (this.iceState.disconnected) {
+      LoggerProxy.logger.log('ReconnectionManager:index#iceDisconnected --> ice has reconnected');
+
+      this.iceState.waitForReconnection.resolve();
+
+      if (this.iceState.timer.clearTimeout) {
+        this.iceState.timer.clearTimeout();
+      }
+
+      this.iceState.waitForReconnection = Promise.resolve();
+      this.iceState.disconnected = false;
+    }
+  }
+
+  /**
+   * @returns {undefined}
+   * @public
+   * @memberof ReconnectionManager
+   */
+  iceDisconnected() {
+    if (!this.iceState.disconnected) {
+      LoggerProxy.logger.log('ReconnectionManager:index#iceDisconnected --> waiting for ice reconnect');
+
+      let resolution;
+
+      this.iceState.waitForReconnection = new Promise((resolve, reject) => {
+        this.iceState.timer = setTimeout(() => {
+          reject(new Error(`ice reconnection did not occur in ${this.iceState.timeoutDuration}ms`));
+        }, this.iceState.timeoutDuration);
+
+        resolution = resolve;
+      });
+
+      this.iceState.waitForReconnection.resolve = resolution;
```

Comment:
Instead of doing it this way, you could strictly save the 'resolve' function and return the promise.
ICE_STATE.DISCONNECTED could attach the `.catch` to the return value of this function.

Reply:
Good idea, implementing this now. I will preserve the promise in the `iceState` object to allow deferred resolution within the class as well.